.teamcity/additionalConfiguration.kt

```diff
@@ -85,4 +83,47 @@ fun Project.deployPlugin() = BuildType {
             gradleWrapperPath = ""
         }
     }
+}.also { buildType(it) }
+
+fun Project.buildWithKotlinMaster(platform: Platform, versionBuild: BuildType) = BuildType {
+    id(BUILD_WITH_KOTLIN_MASTER_ID)
+    this.name = "Build with Kotlin Master (${platform.buildTypeName()})"
+
+    requirements {
+        contains("teamcity.agent.jvm.os.name", platform.teamcityAgentName())
+    }
+    commonConfigure()
+
+    dependsOnSnapshot(versionBuild)
+    params {
+        param(versionSuffixParameter, versionBuild.depParamRefs[versionSuffixParameter].ref)
+        param(teamcitySuffixParameter, versionBuild.depParamRefs[teamcitySuffixParameter].ref)
+    }
+
+    val kotlinVersionParameter = "dep.Kotlin_KotlinPublic_BuildNumber.deployVersion"
+
+    dependsOn(AbsoluteId("Kotlin_KotlinPublic_Artifacts")) {
+        artifacts {
+            buildRule = lastSuccessful()
+            cleanDestination = true
+            artifactRules = "+:maven.zip!**=>artifacts/kotlin"
```

Comment:
Both AFU and Serialization are also copying `+:kotlin-native-prebuilt-linux-x86_64-%dep.Kotlin_KotlinPublic_BuildNumber.deployVersion%.tar.gz!** => artifacts/native`.

Does it make sense here as well (for sample projects)?

Reply:
The unpacked `maven.zip` already contains the K/N distribution, and Gradle gets it from there. It is specified by `-Pkotlin_repo_url=<checkoutDir>/artifacts/kotlin`. I didn't find where and how `artifacts/native` is used, thus dropped it in the last commit.